### PR TITLE
Cloud migrations changes

### PR DESCRIPTION
**Pull request by thekatstevens**
This PR just leaves some breadcrumbs and instructions for users looking to migrate from self-hosted to cloud, based on the decisions made [here](https://docs.google.com/document/d/1APxPoN_TUv4wZY_V92re4mZQisyE5Uj1RKAFsZr_imw/edit#).

**Before merging:**

- [ ] figure out how we want to receive the dataz, and update `migrations.md` with instructions
- [ ] confirm the list of dataz we want the user to generate
- [ ] add some guidance on generating the specific item(s) we want to receive (e.g. how to get the master key, how to export the database, where to find packages, etc.)

### CODE REVIEW

File: docs/octopus-cloud/migrations.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+**Note:** our existing [Migration API](/docs/api-and-integration/migration-api/index.md) is **not supported** for migrations to cloud instances at this time.
+:::
+
+## Before you start
+
+This migration process is intended as a one-off operation to lift and shift everything across to an Octopus Cloud instance. Before you schedule time with our team to do the migration, please keep in mind:
+
+- there will be an outage of 1-2 business days while we migrate your data through to a running cloud instance
+- we will need a number of data files to do the migration on our end, so please schedule some time to export the necessary files and artifacts
+- exports will need to be generated from a **minimum** version of 2018.10, so please make sure you are running this version or higher before exporting the database
```
Comment (thekatstevens) [inline docs/octopus-cloud/migrations.md]: I've just guessed a version of `2018.10` here, as that is our first LTS cut. Happy for this to be changed to whatever though

File: docs/octopus-cloud/migrations.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+:::info
+**Why can't I do a self-service migration at my own pace?**
+We are currently working on making the process more seamless without such a long outage period, and plan to release some tooling to help streamline the data-gathering process sometime around mid-2019. Keep an eye out on our blog for updates on our progress in this area.
```
Comment (thekatstevens) [inline docs/octopus-cloud/migrations.md]: I've kept it vague with `mid-2019` - if that's not vague enough, we can make it even less specific

File: docs/octopus-cloud/migrations.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+:::
+
+## The migration process
+
+1. Book in a mutually convenient time with our team by emailing [support@octopus.com](mailto:support@octopus.com), so you can schedule the outage to a time that suits your needs. Please allow 1-2 business days of downtime
```
Comment (thekatstevens) [inline docs/octopus-cloud/migrations.md]: Currently no appointlet set up for this, so users can just email to book something in, and we handle "the schedule" that way. I don't mind it - it's a little more friction, so only the really tenacious users will take the time - which will help set the tempo we want.

I've also said `1-2 business days` as the expected outage time - is that enough?

File: docs/octopus-cloud/migrations.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+1. Make sure your Octopus Server is running _at least_ version [2018.10](https://octopus.com/downloads/2018.10.0) (or higher), so avoid any complications with the migration process
+1. When you're ready to do the migration, shutdown your Octopus Server and generate the required files (see below), and send these through to us via `<How do we want to receive these files??>`
```
Comment (thekatstevens) [inline docs/octopus-cloud/migrations.md]: Wasn't sure how we are planning to receive the files - `SecureFile` or Dropbox or something? If we can work that out, I'll add some guidance here. It needs to be secure, no matter what we choose

File: docs/octopus-cloud/migrations.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+1. We'll be in contact within 1-2 business days from when you supply the files to us with an update on the progress of the migration to Octopus Cloud
+
+## Required files for migration
+
+- Full Octopus database backup
+- Master key
+- Packages?
+- Artifacts?
+- Task logs / server logs?
+- ???
```
Comment (thekatstevens) [inline docs/octopus-cloud/migrations.md]: Please confirm this list of "things we would need to do a full migration" - once the list is complete, I'll add some guidance on where/how to produce all of these things

File: docs/octopus-cloud/migrations.md (hunk 2 above)
Comment (pawelpabich) [reply docs/octopus-cloud/migrations.md]: in the future :)

**Comment by pawelpabich** [comment]
Looks good to me 👍 . @lawrencewilson @stopthatastronaut  can you help with the list of things we would need for the migration?

File: docs/administration/data/data-migration.md [hunk 6]
```diff
@@ -28,7 +28,7 @@ The data migration tools are not suitable for every imaginable scenario. In thes
 1. For general disaster recovery learn about [backup and restore for your Octopus Server](/docs/administration/data/backup-and-restore.md).
 1. To move your Octopus database to another database server see [moving your database](/docs/administration/managing-infrastructure/moving-your-octopus/move-the-database.md).
 1. To move your Octopus Server and database to another server see [moving your Octopus Server and database](/docs/administration/managing-infrastructure/moving-your-octopus/move-the-database-and-server.md).
-1. To move your entire Octopus Server from a self-hosted installation to Octopus Cloud, contact our [support team](https://octopus.com/support) for help.
+1. To move your entire Octopus Server from a self-hosted installation to Octopus Cloud, see [migrating from self-hosted to Octopus Cloud](/docs/octopus-cloud/migrations.md).
```
Comment (michaelnoonan) [inline docs/administration/data/data-migration.md]: Believe it or not, we just had our first case of "I want to go from Octopus Cloud to self-hosted..." 🤓

File: docs/octopus-cloud/migrations.md (hunk 1 above)
Comment (michaelnoonan) [reply docs/octopus-cloud/migrations.md]: I think this doc should leave some room for us to dictate the version. "You will need to upgrade your self-hosted instance to a specific version in preparation for the migration."

File: docs/octopus-cloud/migrations.md (hunk 3 above)
Comment (michaelnoonan) [reply docs/octopus-cloud/migrations.md]: We can hook up an appointlet easily enough. Alternatively we can leave it open so we can talk with people up front over in support@octopus.com so they can prepare.

File: docs/administration/data/data-migration.md (hunk 6 above)
Comment (thekatstevens) [reply docs/administration/data/data-migration.md]: I'll add it in as a "contact support" type case

File: docs/octopus-cloud/migrations.md (hunk 1 above)
Comment (thekatstevens) [reply docs/octopus-cloud/migrations.md]: I've reworded this to be less specific

File: docs/octopus-cloud/migrations.md (hunk 3 above)
Comment (thekatstevens) [reply docs/octopus-cloud/migrations.md]: I think leaving it as an email process is good - so we can open the conversion 👍

File: docs/octopus-cloud/data.md [hunk 7]
```diff
@@ -10,7 +10,7 @@ We evaluate our backup and recovery strategy frequently to improve resiliency an
 
 ## Importing Data
 
-To import any projects you will need to set them up from scratch, or import them using the [Migration API](/docs/api-and-integration/migration-api/index.md). There are futures plans to provide a migration tool for moving to from an existing on-premise instance into the cloud.
+Due to some key differences between the self-hosted and cloud configurations, it currently **isn't** possible to use the existing [Migration API](/docs/api-and-integration/migration-api/index.md). If you are migrating from an existing self-hosted instance please see [migrating from self-hosted to Octopus Cloud](/docs/octopus-cloud/migrations.md).
```
Comment (wordlee) [inline docs/octopus-cloud/data.md]: To make it super clear, I'd make **isn't** **is not**.

**Comment by wordlee** [comment]
Looks good, Andy. I've submitted a patch for you to review. The edits in the patch are mostly word choices,  and I've tried to remove some duplication in the text.

**Comment by lawrencewilson** [comment]
> Looks good to me 👍 . @lawrencewilson @stopthatastronaut can you help with the list of things we would need for the migration?

@pawelpabich sure no problem I think at this time we just need the standard items like: database export, master key and file backup.

I think customers might encounter issues getting a database backup if they have SQLExpress if memory serves correctly. I'll run through a practice run and see if there is anything specific I can add to that list.

**Comment by thekatstevens** [comment]
@lawrencewilson @pawelpabich I've reworked the wording to leave the process very vague - when customers reach out to book this in, we can then discuss what specific steps are involved, and what files we might need from them. So it's probably still worth having the process written up _somewhere_ as a bit of a songsheet, but not important to expose that to users 😄

**Pull request merged**